### PR TITLE
List changed pages in docs preview

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/10951?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10951/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10951
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

List changed pages in docs preview for convenience.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/workflows/preview-docs.yml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 name: Preview docs
 
 on:
-  pull_request_target:
+  # pull_request_target:
+  pull_request:
```
Comment (harupy) [inline .github/workflows/preview-docs.yml:5]:
```suggestion
  pull_request_target:
```

File: docs/source/python_api/mlflow.transformers.rst [hunk 2]
```diff
@@ -4,4 +4,7 @@ mlflow.transformers
 .. automodule:: mlflow.transformers
     :members:
     :undoc-members:
-    :show-inheritance:
\ No newline at end of file
+    :show-inheritance:
+
+Hello
+-----
```
Comment (harupy) [inline docs/source/python_api/mlflow.transformers.rst:10]:
```suggestion
    :show-inheritance:
```

File: docs/changed_pages.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    SOURCE_REGEX = re.compile(r"<!-- source: (.+) -->")
+    BUILD_DIR = pathlib.Path("build/html")
+    changed_files = fetch_changed_files(pr_num)
+    changed_pages: set[str] = set()
+    for p in BUILD_DIR.rglob("**/*.html"):
+        if m := SOURCE_REGEX.search(p.read_text()):
+            source = m.group(1)
+            if source in changed_files:
+                changed_pages.add(p.relative_to(BUILD_DIR))
```
Comment (harupy) [inline docs/changed_pages.py:36]: This only detects .rst file updates and doesn't detect .py file updates. Extra effort is required to support that.

File: docs/source/llms/transformers/tutorials/audio-transcription/whisper.ipynb [hunk 4]
```diff
@@ -6,7 +6,7 @@
    "source": [
     "## Introduction to MLflow and OpenAI's Whisper\n",
     "\n",
-    "Discover the integration of [OpenAI's Whisper](https://huggingface.co/openai), an [ASR system](https://en.wikipedia.org/wiki/Speech_recognition), with MLflow in this tutorial.\n",
+    "Discover THE integration of [OpenAI's Whisper](https://huggingface.co/openai), an [ASR system](https://en.wikipedia.org/wiki/Speech_recognition), with MLflow in this tutorial.\n",
```
Comment (harupy) [inline docs/source/llms/transformers/tutorials/audio-transcription/whisper.ipynb:9]: Change for testing, will be reverted before merge

**Comment by harupy** [comment]
Test: https://output.circle-artifacts.com/output/job/2e843760-f777-4ead-b169-1f5419d57751/artifacts/0/docs/build/html/diff.html

**Review by serena-ruan — APPROVED** [review]
LGTM!! Thanks so much this is super helpful!

**Review by BenWilson2 — APPROVED** [review]
+1 this is great

**Pull request merged**